generator/docs/GenerationTypes.md

```diff
@@ -27,13 +27,30 @@ Examples:
 
-| Constraint | Emitted valid data |Emitted violating data |
+| Data type | where there are no constraints for the field <br /> _(or only `ofType` and/or `not(is null)`)_ | where constraints exist for the field <br /> _(beyond `ofType` and `not(is null)`)_ |
 | ---- | ---- | ---- |
-| `Field 1 > 10 AND Field 1 < 20` | 11, 19 | 10, 20 |
-| `Field 1 in set [A, B, C]` | A, B, C* | `null` |
-
-\* This may change in future.
\ No newline at end of file
+| **string** | "Lorem Ipsum" | a string as short as possible given the intersection of other constraints <br /> a string as long as possible given the intersection of other constraints <br /> If a containsRegex or matchingRegex is present, use this to inform the characters that are included in the shortest/longest strings. Otherwise repeat a `<space>` character for the contents of both strings <br /> |
+| **numeric (integer values)** <br /> where all present numeric constraints (if any) contain whole numbers and no decimal points | -2147483648 <br /> 0 <br /> 2147483646 | _A unique set of:_ <br /> the lowest possible value or -2147483648 <br /> 0 (if permitted) <br /> the highest possible value or 2147483646 |
+| **numeric (decimal values)** <br /> where some numeric constraints contain a decimal point (even if the fraction is 0, e.g. 10.0) | -2147483648 <br /> 0 <br /> 2147483646 | _A unique set of:_ <br /> the 2 lowest possible values or **??** [see #529](https://github.com/ScottLogic/datahelix/issues/529) <br /> 0 (if permitted) <br /> the 2 highest possible values or 1.797693134862315699E+308 & 1.797693134862315700E+308 |
+| **temporal** | 0000-01-01T00:00:00.000 <br /> 9999-12-31T23:59:59.999 | _A unique set of:_ <br /> the earliest possible date or 0000-01-01T00:00:00.000 <br /> the latest possible date or 9999-12-31T:23:59:59.999 |
+| **string** (`aValid ISIN`) | "GB0000000009" <br /> "GB00JJJJJJ45" <br /> "US0000000002" | NA - this constraint has no other constraints that can be used in conjunction with it - see [#488](https://github.com/ScottLogic/datahelix/issues/488) |
+
+### Conditional constraints
+Any value/s mentioned within an `if` or `anyOf` will be considered interesting whether the field is constrained or not. Consider the following profile:
+```
+field1 greaterThanOrEqual 0
+field1 lessThan 10
+field1 ofType numeric
+
+if field1 equalTo 5 then
+  field2 equalTo "a"
+```
+
+Given the table above the interesting values for `field1` would be `0` & `9`. To ensure the consequence of the `if` constraint is met (`field2 equalTo "a"`), at least once, `5` is included as an interesting value. As such the above profile would emit `0`, `5` & `9` for `field1`.
+
+### Sets
+All values of the `inSet` constraint is used (the single value in an `equalTo` constraint) are considered interesting. There is no discrimination over the values emitted for these sets.
```

Comment:
Does this include the implicit null?

Reply:
Updated to clarify that `null` is an interesting 'value'